src/NzbDrone.Core/Profiles/Releases/PreferredWordService.cs

```diff
@@ -82,5 +84,41 @@ public List<string> GetMatchingPreferredWords(Series series, string title)
                                 .Select(p => p.Key)
                                 .ToList();
         }
-    }
+
+        public Dictionary<string, List<string>> GetMatchingPreferredWordsGroupByProfile(Series series, string title)
```

Comment:
There should be a single function that returns all preferred words, both ungrouped and grouped, no point in making multiple DB calls.

Reply:
I removed the original method, and refactored the callers to use the new Dictionary return.
I used the empty string ("") as the key for the "All Matched Words" bucket, since a `{Preferred Words:}` isn't valid and won't be matched.